python/ray/tune/syncer.py

```diff
@@ -448,3 +449,32 @@ def on_trial_complete(self, iteration: int, trials: List["Trial"],
     def on_checkpoint(self, iteration: int, trials: List["Trial"],
                       trial: "Trial", checkpoint: Checkpoint, **info):
         self._sync_trial_checkpoint(trial, checkpoint)
+
+
+def detect_sync_to_driver(
+        sync_to_driver: Union[None, bool, Type],
+        cluster_config_file: str = "~/ray_bootstrap_config.yaml"):
```

Comment:
can we log a couple debug statements in this class?

this would help identify faulty resolution issues in the future